contracts/libraries/OracleLibrary.sol

```diff
@@ -57,4 +57,21 @@ library OracleLibrary {
                 : FullMath.mulDiv(1 << 128, baseAmount, ratioX128);
         }
     }
+
+    /// @notice Given a pool, it returns the number of seconds ago of the oldest stored observation
+    /// @param pool Address of Uniswap V3 pool that we want to observe
+    /// @return The number of seconds ago of the oldest observation stored for the pool
+    function getOldestObservationSecondsAgo(address pool) internal view returns (uint32) {
+        (, , uint16 observationIndex, uint16 observationCardinality, , , ) = IUniswapV3Pool(pool).slot0();
+        require(observationCardinality > 0, 'Pool not initialized');
+
+        (uint32 observationTimestamp, , , bool initialized) =
+            IUniswapV3Pool(pool).observations((observationIndex + 1) % observationCardinality);
+
+        if (!initialized) {
+            (observationTimestamp, , , ) = IUniswapV3Pool(pool).observations(0);
+        }
+
+        return uint32(block.timestamp) - observationTimestamp;
```

Comment:
revisiting this after looking at the tests - in core we take in `block.timestamp` as an argument to libraries (e.g. https://github.com/Uniswap/uniswap-v3-core/blob/main/contracts/libraries/Oracle.sol#L147)

doing this here would increase gas costs a bit and probably be a little confusing to callers, so i'm not sure it's worth it (the big benefit is ease of unit testing, both for use and dev users), but something to consider/be aware of.

@moodysalem might have thoughts here

Reply:
@moodysalem after chatting with @hensha256 we think that this is the preferred API for callers, so we're going to leave as is and try to get our tests to a spot where they feel clear and thorough - any strong objection to that?